### PR TITLE
Allow users to pm themselves while /blockpms is active

### PR DESCRIPTION
**Pull request by sirKiraUzumaki**
Suggestion: https://www.smogon.com/forums/threads/allow-users-to-chat-themselves-with-blockpms-active.3672227/

### CODE REVIEW

File: server/chat.ts [hunk 1]
```diff
@@ -984,7 +984,7 @@ export class CommandContext extends MessageContext {
 				}
 				if (targetUser.settings.blockPMs &&
 					(targetUser.settings.blockPMs === true || !Users.globalAuth.atLeast(user, targetUser.settings.blockPMs)) &&
-					!user.can('lock')) {
+					!user.can('lock') && toID(targetUser) != toID(user)) {
```
Comment (mia-pi-git) [inline server/chat.ts:987]:
```suggestion
					!user.can('lock') && targetUser.id != user.id) {
```

Comment (mia-pi-git) [reply server/chat.ts:987]: (if you use toID on a user object, it returns user.id, so it's just cleaner this way)

Comment (AnnikaCodes) [reply server/chat.ts]: This should also be `!==` not `!=`.

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**